extensions/Worker.Extensions.CosmosDB/src/CosmosDBConverter.cs

```diff
@@ -0,0 +1,188 @@
+            {
+                throw new InvalidOperationException("The 'Id' and 'PartitionKey' properties of a CosmosDB single-item input binding cannot be null or empty.");
+            }
+
+            ItemResponse<T> item = await container.ReadItemAsync<T>(cosmosAttribute.Id, new PartitionKey(cosmosAttribute.PartitionKey));
+
+            if (item is null || item?.StatusCode is not System.Net.HttpStatusCode.OK || item.Resource is null)
+            {
+                throw new InvalidOperationException($"Unable to retrieve document with ID '{cosmosAttribute.Id}' and PartitionKey '{cosmosAttribute.PartitionKey}'");
+            }
+
+            return item.Resource;
+        }
+
+        private async Task<object> CreatePOCOCollectionAsync<T>(Container container, CosmosDBInputAttribute cosmosAttribute)
+        {
+            QueryDefinition queryDefinition = null!;
+            if (!String.IsNullOrEmpty(cosmosAttribute.SqlQuery))
+            {
+                queryDefinition = new QueryDefinition(cosmosAttribute.SqlQuery);
+                if (cosmosAttribute.SqlQueryParameters?.Count() > 0)
+                {
+                    foreach (var parameter in cosmosAttribute.SqlQueryParameters)
+                    {
+                        queryDefinition.WithParameter(parameter.Key, parameter.Value.ToString());
+                    }
+                }
+            }
+
+            QueryRequestOptions queryRequestOptions = new();
+            if (!String.IsNullOrEmpty(cosmosAttribute.PartitionKey))
+            {
+                var partitionKey =  new PartitionKey(cosmosAttribute.PartitionKey);
+                queryRequestOptions = new() { PartitionKey = partitionKey };
+            }
+
+            using (var iterator = container.GetItemQueryIterator<T>(queryDefinition: queryDefinition, requestOptions: queryRequestOptions))
+            {
+                if (iterator is null)
+                {
+                    throw new InvalidOperationException($"Unable to retrieve documents for container '{container.Id}'.");
+                }
+
+                return await ExtractCosmosDocumentsAsync(iterator);
+            }
+        }
+
+        private async Task<IList<T>> ExtractCosmosDocumentsAsync<T>(FeedIterator<T> iterator)
+        {
+            var documentList = new List<T>();
+            while (iterator.HasMoreResults)
+            {
+                FeedResponse<T> response = await iterator.ReadNextAsync();
+                documentList.AddRange(response.Resource);
+            }
+            return documentList;
+        }
+
+        private T CreateCosmosClient<T>(CosmosDBInputAttribute cosmosAttribute)
+        {
+            var cosmosDBOptions = _cosmosOptions.Get(cosmosAttribute?.Connection);
+            CosmosClient cosmosClient = cosmosDBOptions.GetClient(cosmosAttribute?.Connection!, cosmosAttribute?.PreferredLocations!);
```

Comment:
We already get the cosmos options based on `Connection`, passing connection in to `GetClient` seems redundant and also incorrect. It looks to me you shouldn't be able to get options with connection "A" then call GetClient with connection "B" - that would provide an incorrect client.

Removing `Connection` from the `GetClient` call would fix this.

Reply:
connection is being passed in just for the cache which we decided to remove anyway in favor of ` internal CosmosClient? Client { get; set; }` like we do in blob - seems like we never got round to removing that so I'll fix this now (and remove use of connection here as it will no longer be needed)